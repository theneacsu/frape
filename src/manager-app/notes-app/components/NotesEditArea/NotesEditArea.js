import React from 'react'
import { connect } from 'react-redux'
import NotesForm from '../NotesForm/NotesForm'
import {
  startEditNote,
  startRemoveNote
} from '../../../../actions/notes-app/notes'
import styles from './NotesEditArea.module.css'

export const NotesEditArea = props => (
  <div className={styles.div}>
    <NotesForm
      purpose="edit"
      note={props.note}
      onSubmit={note => {
        props.startEditNote(props.note.id, note)
        props.history.push('/notes')
      }}
      onRemove={() => {
        props.startRemoveNote(props.note.id)
        props.history.push('/notes')
      }}
    />
  </div>
)

const mapStateToProps = (state, props) => ({
  note: state.notes.find(note => note.id === props.match.params.id)
})

const mapDispatchToProps = dispatch => ({
  startEditNote: (id, updates) => dispatch(startEditNote(id, updates)),
  startRemoveNote: id => dispatch(startRemoveNote(id))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotesEditArea)
