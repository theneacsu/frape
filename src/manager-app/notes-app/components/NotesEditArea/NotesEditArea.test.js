import React from 'react'
import { shallow } from 'enzyme'
import { NotesEditArea } from './NotesEditArea'

let note, startEditNoteSpy, startRemoveNoteSpy, history, wrapper

beforeEach(() => {
  note = {
    id: '123abc',
    title: 'I got my mind set on you',
    body: 'old music'
  }
  startEditNoteSpy = jest.fn()
  startRemoveNoteSpy = jest.fn()
  history = {
    push: jest.fn()
  }
  wrapper = shallow(
    <NotesEditArea
      history={history}
      note={note}
      startEditNote={startEditNoteSpy}
      startRemoveNote={startRemoveNoteSpy}
    />
  )
})

test('Should render the NotesEditArea component correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should call startEditNote', () => {
  const noteUpdate = {
    text: 'bla bla'
  }
  wrapper.find('NotesForm').prop('onSubmit')(noteUpdate)
  expect(startEditNoteSpy).toHaveBeenLastCalledWith(note.id, noteUpdate)
  expect(history.push).toHaveBeenLastCalledWith('/notes')
})

test('Should call startRemoveNote', () => {
  wrapper.find('NotesForm').prop('onRemove')()
  expect(startRemoveNoteSpy).toHaveBeenLastCalledWith(note.id)
  expect(history.push).toHaveBeenLastCalledWith('/notes')
})
