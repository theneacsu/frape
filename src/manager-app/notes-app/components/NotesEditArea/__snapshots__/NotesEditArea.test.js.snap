// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the NotesEditArea component correctly 1`] = `
<div
  className="div"
>
  <NotesForm
    note={
      Object {
        "body": "old music",
        "id": "123abc",
        "title": "I got my mind set on you",
      }
    }
    onRemove={[Function]}
    onSubmit={[Function]}
    purpose="edit"
  />
</div>
`;
